packages/jsii-pacmak/lib/targets/go/types/class.ts

```diff
@@ -165,3 +168,24 @@ export class ClassMethod implements GoTypeMember {
     );
   }
 }
+
+export class StaticMethod extends ClassMethod {
+  public constructor(
+    public readonly parent: GoClass,
+    public readonly method: Method,
+  ) {
+    super(parent, method);
+  }
+
+  public emit({ code }: EmitContext) {
+    const name = `${this.parent.name}_${this.name}`;
+    const returnTypeString = this.reference?.void ? '' : ` ${this.returnType}`;
+
+    code.openBlock(`func ${name}(${this.paramString()})${returnTypeString}`);
+
+    this.runtimeCall.emit(code);
+
+    code.closeBlock();
+    code.line();
+  }
```

Comment:
It seems the difference between this override and the implementation in the base class (`ClassMethod`) is pretty minimal. I wonder if it'd not be "safer" to branch in `ClassMethod` based on `this.method.abstract` instead of sub-classing here?